### PR TITLE
Fix configuration caching

### PR DESCRIPTION
**Pull request by zangell44**
Resolves a bug with configuration caching introduced in #6959

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

File: src/prefect/orion/database/query_components.py [hunk 1]
```diff
@@ -396,10 +396,15 @@ async def read_configuration(
...
             result = await session.execute(query)
-            value = result.scalar()
-            if value is not None:
-                self.CONFIGURATION_CACHE[key] = value
-            return value
+            configuration = result.scalar()
+            if configuration is not None:
+                self.CONFIGURATION_CACHE[key] = configuration.value
+                return configuration.value
+            return configuration
```
Comment (jakekaplan) [inline src/prefect/orion/database/query_components.py:403]: should this be `configuration.value`?

Comment (zangell44) [reply src/prefect/orion/database/query_components.py:403]: in this case `configuration` is `None`

Comment (jakekaplan) [reply src/prefect/orion/database/query_components.py:403]: ah I see

**Review by jakekaplan — APPROVED** [review]
LGTM

Not sure if someone more contextually familiar with configurations wants to take a look 😅

**Pull request merged**